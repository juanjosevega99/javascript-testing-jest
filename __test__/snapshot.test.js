import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe("it is time for snapshots", () => {
  test("Snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test("fail snapshot", () => {
    const user = {
      id: Math.floor(Math.random() * 100),
      name: "Juan Jose Vega",
      headline: "Full Stack Developer",
      createdAt: new Date(),
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number),
      createdAt: expect.any(Date),
    });
  });

  test("we have an exception", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Juan Jose Vega",
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});
