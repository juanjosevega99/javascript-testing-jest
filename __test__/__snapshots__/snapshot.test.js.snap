// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it is time for snapshots Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
